community/kernel/src/main/java/org/neo4j/kernel/extension/dependency/DeleteStoresFromOtherLabelScanStoreProviders.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.extension.dependency;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+
+import org.neo4j.graphdb.DependencyResolver;
+import org.neo4j.kernel.impl.api.scan.LabelScanStoreProvider;
+
+/**
+ * Given an already selected {@link LabelScanStoreProvider}, tell all the other {@link LabelScanStoreProvider}
+ * to {@link LabelScanStoreProvider#drop() drop} their stores, so that there's only one authoritative provider.
+ */
+public class DeleteStoresFromOtherLabelScanStoreProviders implements DependencyResolver.SelectionStrategy
```

Comment:
I would say if you have some time now - we do that now and i would prefer that, otherwise lets add a tech debt card and fix it during next tech debt cleanup work.